src/Http/Request.php

```diff
@@ -150,6 +266,50 @@ public function getHttpFoundationRequest()
         $post    = $this->getPost();
         $cookies = $this->getCookies();
 
-        return new HttpFoundationRequest($query, $post, [], $cookies, [], $this->params, $this->stdin);
+        return new HttpFoundationRequest($query, $post, [], $cookies, $this->uploadedFiles, $this->params, $this->stdin);
+    }
+
+    /**
+     * Add a file to the $files array
+     */
+    private function addFile(array &$files, string $fieldName, string $filename, string $tmpPath, string $mimeType, bool $err): void
+    {
+        $data = [
+            'type' => $mimeType ?: 'application/octet-stream',
+            'name' => $filename,
+            'tmp_name' => $tmpPath,
+            'error' => $err ? UPLOAD_ERR_CANT_WRITE : UPLOAD_ERR_OK,
+            'size' => filesize($tmpPath),
+        ];
+
+        if (!preg_match('|([^\[]+)(?:\[([^\]]+)\])*(\[\])?|', $fieldName, $matches)) {
```

Comment:
This regex is not good enough. It does not pick up "foo" in `two[foo][bar][]`. 

Any ideas for improved regex?

https://regex101.com/r/yyQorq/2